languages/purescript/reference/errors/InvalidInstanceHead.md

```diff
@@ -0,0 +1,131 @@
+# `InvalidInstanceHead` Error
+
+## Example
+
+As of 0.11.6, the `InvalidInstanceHead` error has been added to the
+purescript compiler. The error can be produced using the following:
+
+```purescript
+module InvalidInstanceHead where
+
+class ClassWithInvalidInstanceHead (x :: # Type)
+
+instance FailingExample :: ClassWithInvalidInstanceHead ()
+...
+```
+
+## Cause
+
+Normally, when you see this error, it's because you're trying to write an
+`instance` of a given `class` that pattern matches on some **row type**. The
+issue here is that there isn't really a sensible way to match on row. There
+are, however, ways to get around this limitation.
+
+## Fix
+
+With the arrival of [`purescript-typelevel-prelude`](https://github.com/purescript/purescript-typelevel-prelude)
+2.3.0 came the introduction of `RowToList` and its related classes. The
+basic idea is that you can convert a row type to a type-level cons list of its
+entries. For example:
+
+```purescript
+--        ROW              LIST
+
+RowToList ()               Nil
+RowToList (a :: A)         (Cons "a" A Nil)
+RowToList (a :: A, b :: B) (Cons "a" A (Cons "b" B Nil))
+```
+
+The `RowToList` constraint allows us to derive, from a row type, the `RowList`
+that complements it. The keys of the `RowList` are alphabetically ordered, which
+allows us to manipulate row types in a deterministic way. If your instance heads
+pattern match on `RowList` instead of `# Type`, this error goes away.
+
+## Worked Example
```

Comment:
*Working

Reply:
It's `Worked` as in they work through it with you. Maybe I should make it `Worked Working Example` :stuck_out_tongue: